src/Compilers/CSharp/Portable/CodeGen/CodeGenerator.cs

```diff
@@ -102,12 +131,17 @@ private enum IndirectReturnState : byte
 
             _boundBody = Optimizer.Optimize(
                 boundBody, 
-                debugFriendly: _optimizations != OptimizationLevel.Release, 
+                debugFriendly: _ilEmitStyle != ILEmitStyle.Release, 
                 stackLocals: out _stackLocals);
 
             _methodBodySyntaxOpt = (method as SourceMethodSymbol)?.BodySyntax;
         }
 
+        private bool IsDebugPlus()
+        {
+            return this._module.Compilation.Options.DebugPlusMode;
```

Comment:
Should we be consistent with the way we handle OptimizationLevel? I.e. pass it in rather than rely on the value stored in compilation?